internal/persistence/sql/persister.go

```diff
@@ -28,121 +29,101 @@ import (
 
 type (
 	Persister struct {
-		conn       *pop.Connection
-		mb         *popx.MigrationBox
-		namespaces namespace.Manager
-		l          *logrusx.Logger
-		dsn        string
-		tracer     *tracing.Tracer
+		conn             *pop.Connection
+		d                dependencies
+		nid              uuid.UUID
+		namespaceIDCache *ristretto.Cache
 	}
 	internalPagination struct {
 		Page, PerPage int
 	}
-	contextKeys string
+	dependencies interface {
+		config.Provider
+		x.LoggerProvider
+
+		Tracer() *tracing.Tracer
+		PopConnection() (*pop.Connection, error)
+	}
 )
 
 const (
 	defaultPageSize int = 100
 )
 
 var (
-	//go:embed migrations/*.sql
+	//go:embed migrations/sql/*.sql
 	migrations embed.FS
 
-	//go:embed namespace_migrations/*.sql
-	namespaceMigrations embed.FS
+	////go:embed namespace_migrations/*.sql
+	//namespaceMigrations embed.FS
 
 	_ persistence.Persister = &Persister{}
 )
 
-func NewPersister(dsn string, l *logrusx.Logger, namespaces namespace.Manager, tracer *tracing.Tracer) (*Persister, error) {
-	pop.SetLogger(l.PopLogger)
-
-	p := &Persister{
-		namespaces: namespaces,
-		l:          l,
-		dsn:        dsn,
-		tracer:     tracer,
-	}
+func NewPersister(reg dependencies, nid uuid.UUID) (*Persister, error) {
+	//pop.SetLogger(reg.Logger().PopLogger)
 
-	var err error
-	p.conn, err = p.newConnection(nil)
+	pop.Debug = true
+	conn, err := reg.PopConnection()
 	if err != nil {
 		return nil, err
 	}
 
-	return p, nil
-}
-
-func (p *Persister) newConnection(options map[string]string) (c *pop.Connection, err error) {
-	var opts []instrumentedsql.Opt
-	if p.tracer.IsLoaded() {
-		opts = []instrumentedsql.Opt{
-			instrumentedsql.WithTracer(opentracing.NewTracer(true)),
-			instrumentedsql.WithOmitArgs(),
-		}
+	namespaceIDCache, err := ristretto.NewCache(&ristretto.Config{
+		// TODO probably make this configurable
+		NumCounters: 10 * 1000,
+		MaxCost:     1000,
```

Comment:
This will hold 1000 cost units - the num counters is correct (but use a const and just do 10*const for numcounters).

Since it appears that you want to store 1000 namespace objects, you will need to disable the internal cost calculator so that 1 namespace really has 1 space. Otherwise the key is included in the cost in ristrettoy:

```suggestion
		MaxCost:     1000,
		IgnoreInternalCost: true,
```